sage/example-transition/src/transition/hero_jam.rs

```diff
@@ -0,0 +1,266 @@
+		match transition_id {
+			HeroAction::Create => {
+				let asset_id = ChainHandler::get_current_block_number().saturated_into::<AssetId>();
+				let asset = HeroJamAsset {
+					id: asset_id,
+					asset_type: AssetType::None,
+					asset_subtype: AssetSubType::None,
+					energy: 100,
+					fatigue: 0,
+					state_type: StateType::None,
+					state_sub_type: 0,
+					state_sub_value: 0,
+					state_change_block_number: 0_u32.saturated_into::<BlockNumber>(),
+					balance: 10,
+				};
+				Ok(vec![TransitionOutput::Minted(Asset::from(asset))])
+			},
+			HeroAction::Sleep(sleep_time) => {
+				match AssetHandler::get_asset(&asset_ids[0])
+					.map_err(|_| TransitionError::Transition { error: 0 })?
+					.asset_variant
+				{
+					AssetVariant::HeroJam(mut asset) => {
+						asset.state_type = StateType::Sleep;
+						asset.state_sub_type = SleepType::Normal as u8;
+						asset.state_sub_value = *sleep_time as u8;
+						asset.state_change_block_number =
+							Self::get_block_time_for_action(sleep_time);
+
+						Ok(vec![TransitionOutput::Mutated(asset_ids[0], Asset::from(asset))])
+					},
+				}
+			},
+			HeroAction::Work(work_time, work_type) => {
+				match AssetHandler::get_asset(&asset_ids[0])
+					.map_err(|_| TransitionError::Transition { error: 0 })?
+					.asset_variant
+				{
+					AssetVariant::HeroJam(mut asset) => {
+						let fatigue = Self::get_resource_fatigue(
+							asset.state_type,
+							asset.state_sub_type,
+							asset.state_sub_value,
+						);
+						let energy = Self::get_resource_energy(
+							asset.state_type,
+							asset.state_sub_type,
+							asset.state_sub_value,
+						);
+
+						if work_type == &WorkType::Hunt {
+							asset.balance = asset.balance.saturating_add(10);
+						}
+
+						asset.fatigue =
+							(asset.fatigue as i32).saturating_add(fatigue).clamp(0, 255) as u8;
+						asset.energy =
+							(asset.energy as i32).saturating_add(energy).clamp(0, 255) as u8;
+
+						asset.state_type = StateType::Work;
+						asset.state_sub_type = *work_type as u8;
+						asset.state_sub_value = *work_time as u8;
+						asset.state_change_block_number =
+							Self::get_block_time_for_action(work_time);
+
+						Ok(vec![TransitionOutput::Mutated(asset_ids[0], Asset::from(asset))])
+					},
+				}
+			},
+			_ => Ok(vec![]),
+		}
+	}
+
+	fn get_block_time_for_action(action_time: &ActionTime) -> BlockNumber {
+		let current_block = ChainHandler::get_current_block_number();
+		let block_time = action_time.get_block_time_from();
+
+		current_block.saturating_add(block_time.saturated_into())
+	}
+
+	fn get_resource_fatigue(state_type: StateType, state_sub_type: u8, action_time: u8) -> i32 {
+		match state_type {
+			StateType::None => 0,
+			StateType::Sleep => match SleepType::from(state_sub_type) {
+				SleepType::Normal => -((action_time as i32) * (action_time as i32) * 11),
+				_ => 0,
+			},
+			StateType::Work => (action_time as i32) * 8,
```

Comment:
Never use signed types if not necessary. You can omit the type annotations on the left hand side if you put  `* 8u32*